internal/errors/check.go

```diff
@@ -0,0 +1,35 @@
+package errors
+
+// Check returns retry mode for err.
+func Check(err error) (
+	statusCode int32,
+	operationStatus OperationStatus,
+	backoff BackoffType,
+	deleteSession bool,
+) {
+	var te *transportError
+	var oe *operationError
+	var re *RetryableError
+	switch {
+	case As(err, &te):
+		return int32(te.code),
+			te.OperationStatus(),
+			te.BackoffType(),
+			te.MustDeleteSession()
+	case As(err, &oe):
+		return int32(oe.code),
+			oe.OperationStatus(),
+			oe.BackoffType(),
+			oe.MustDeleteSession()
+	case As(err, &re):
+		return -1,
+			OperationNotFinished,
+			re.BackoffType,
+			re.MustDeleteSession
+	default:
+		return -1,
+			OperationFinished, // it's finish, not need any retry attempts
```

Comment:
it's finish**ed**, not need any retry attempts

Reply:
ack